feature_engine/selection/drop_duplicate_features.py

```diff
@@ -0,0 +1,76 @@
+import itertools
+from sklearn.base import TransformerMixin, BaseEstimator
+from sklearn.utils.validation import check_is_fitted
+from feature_engine.dataframe_checks import _is_dataframe, _check_input_matches_training_df
+
+
+class DropDuplicateFeatures(BaseEstimator, TransformerMixin):
+    """
+    Drop duplicate features from a dataframe. Duplicate features are those set of features which show same value across
+    all observations.
+    """
+
+    def __init__(self):
+        pass
+
+    def fit(self, X, y=None):
+
+        """
+        Find duplicated features.
+
+        Parameters
+        ----------
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe.
+
+        y: None
+            y is not needed for this transformer. You can pass y or None.
+
+
+        Attributes
+        ----------
+
+        duplicate_feature_dict_: dict
+            The dictionary where keys are features and values are duplicate features.
+        """
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        duplicated_features_list = []
+
+        # create column pairs
+        col_pair = list(itertools.combinations(X.columns, 2))
+
+        # create a dictionary of features with values as list of duplicate features
+        self.duplicate_feature_dict_ = {col: [] for col in X.columns}
+
+        for feat_1, feat_2 in col_pair:
+            if feat_1 not in duplicated_features_list and X[feat_1].equals(X[feat_2]):
+                self.duplicate_feature_dict_[feat_1].append(feat_2)
```

Comment:
After some research / thinking this is a template of what the code could look like:

```
# list of lists or array of arrays: needs some thinking / proper design
duplicated_feat_pairs = []

# create an empty set to collect features
# that were found to be duplicated
_duplicated_feat = ()

# create set of examined features
_examined_features = ()

# pick first feature to compare
for feature in X.columns:
    
     # append so we can remove when we create the combinations
    _examined_features.append(feature)
    
    if feature not in _duplicated_feat:
        
        # remove features being examined, already examined or duplicated features already identified
        remaining_cols = X.columns.drop(list(_duplicated_feat)+list(_examined_features))
        #not too sure we need to convert the sets to lists to be dropped, can you check?
                
        # create combinations of that feature with the others:
        for f1, f2 in itertools.product([feature], remaining_cols):
            
            if data[f1].equals(data[f2]):
                    
                add it to the array or list of lists.
```

A few things to notice:

How to create the array or list of lists needs some thinking. The result could be an empty array if no duplicated features are present in the dataframe. Or alternatively it could contain 1 or more lists / rows with different lenghts. Could I leave the design of this to you?

Another thing I would like to know is whether using itertools as we do here, would be faster than the code suggested in the issue. Could you also check?